### PR TITLE
Fix capsule sync scenario failing with storage exhaust and code improvements

### PR DESCRIPTION
**Pull request by jyejare**
Fixes issue #137

### CODE REVIEW

**Comment by jyejare** [comment]
Test Results post fix:

Scenario : Scenario_capsule_sync
```
# py.test -s -v -m pre_upgrade upgrade_tests/test_scenarios/test_capsule.py::Scenario_capsule_sync
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.11, pytest-3.3.0, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python
cachedir: .cache
rootdir: /home/jitendrayejare2/Desktop/RedHat/sat6-upgrade/satellite6-upgrade, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.3, cov-2.4.0
collected 2 items                                                                                                                                                                                                 

PASSED                                                                                            [100%]

=============================================================================================== 1 tests deselected ================================================================================================
==================================================================================== 1 passed, 1 deselected in 1273.63 seconds ====================================================================================


# py.test -s -v -m post_upgrade upgrade_tests/test_scenarios/test_capsule.py::Scenario_capsule_sync
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.11, pytest-3.3.0, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python
cachedir: .cache
rootdir: /home/jitendrayejare2/Desktop/RedHat/sat6-upgrade/satellite6-upgrade, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.3, cov-2.4.0
collected 2 items                                                                                                                                                                                                 

PASSED                                                                                            [100%]

=============================================================================================== 1 tests deselected ================================================================================================
==================================================================================== 1 passed, 1 deselected in 107.35 seconds =====================================================================================
```

Scenario : Scenario_capsule_sync_2
```
# py.test -s -v upgrade_tests/test_scenarios/test_capsule.py::Scenario_capsule_sync_2
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.11, pytest-3.3.0, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python
cachedir: .cache
rootdir: /home/jitendrayejare2/Desktop/RedHat/sat6-upgrade/satellite6-upgrade, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.3, cov-2.4.0
collected 1 item                                                                                                                                                                                                  

PASSED                                                                                        [100%]

============================================================================================ 1 passed in 3408.85 seconds ============================================================================================
```

**Comment by jyejare** [comment]
The capsule storage after running both tests is not exhausted:
screenshot from 2018-02-06 15-46-01

File: upgrade_tests/test_scenarios/test_capsule.py [hunk 1]
```diff
@@ -137,25 +137,21 @@ def test_post_user_scenario_capsule_sync(self):
...
-        )
         cap_data = hammer.hammer('capsule list')
-        cap_id = hammer.get_attribute_value(cap_data, cap_host, 'id')
-        cap_info = {'id': cap_id, 'name': cap_host}
+        cap_id = hammer.get_attribute_value(cap_data, self.cap_host, 'id')
         org_data = hammer.hammer('organization list')
         org_name = hammer.get_attribute_value(
             org_data, int(self.org_id), 'name')
-        print hammer.sync_capsule_content(cap_info, async=False)
+        print hammer.hammer(
```
Comment (jyejare) [inline upgrade_tests/test_scenarios/test_capsule.py]: The main change is here to fix the issue,

We will now sync environments which were already selected in the capsule. Where before fix we were selecting all the available environments, adding them to satellite and syncing.

File: upgrade_tests/test_scenarios/test_capsule.py [hunk 2]
```diff
@@ -253,22 +249,18 @@ def test_post_user_scenario_capsule_sync_2(self):
...
```
Comment (jyejare) [inline upgrade_tests/test_scenarios/test_capsule.py]: The same main change is here as mentioned in above comment

File: upgrade_tests/test_scenarios/test_capsule.py (hunk 1 above)
Comment (sghai) [reply upgrade_tests/test_scenarios/test_capsule.py]: ok.. just to clear.. sync_capsule_content fn actually sync all environments to capsule which is not a ideal solution. we should sync only the one being added to capsule.

**Pull request merged**